src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_VALID_LOGIN = "Yaroslav";
+    private static final String DEFAULT_VALID_PASSWORD = "123456";
+    private static final int DEFAULT_VALID_AGE = 18;
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_loginNull_NotOk() {
+        User user = new User(null, DEFAULT_VALID_PASSWORD, DEFAULT_VALID_AGE);
+        checkRegistrationException(user);
+    }
+
+    @Test
+    void register_passwordNull_NotOk() {
+        User user = new User(DEFAULT_VALID_LOGIN, null, DEFAULT_VALID_AGE);
+        checkRegistrationException(user);
+    }
+
+    @Test
+    void register_ageNull_NotOk() {
+        User user = new User(DEFAULT_VALID_LOGIN, DEFAULT_VALID_PASSWORD, null);
+        checkRegistrationException(user);
+    }
+
+    @Test
+    void register_ageLess18_NotOk() {
+        User user = new User(DEFAULT_VALID_LOGIN, DEFAULT_VALID_PASSWORD, 5);
+        checkRegistrationException(user);
+    }
+
+    @Test
+    void register_addUserWithTheSameLogin_NotOk() {
+        User user1 = new User(DEFAULT_VALID_LOGIN,
+                DEFAULT_VALID_PASSWORD, DEFAULT_VALID_AGE);
+        User user2 = new User(DEFAULT_VALID_LOGIN,
+                DEFAULT_VALID_PASSWORD + "23", DEFAULT_VALID_AGE + 5);
+        registrationService.register(user1);
+        checkRegistrationException(user2);
+    }
+
+    @Test
+    void register_ageNegativeNumber_NotOk() {
+        User user = new User(DEFAULT_VALID_LOGIN, DEFAULT_VALID_PASSWORD, -10);
+        checkRegistrationException(user);
+    }
+
+    @Test
+    void register_ageMoreThen18_Ok() {
+        User expected = new User(DEFAULT_VALID_LOGIN, DEFAULT_VALID_PASSWORD, 95);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
```

Comment:
Here we need to add more checks to see if the user ID is valid, and if there is a user we added in our list.

Reply:
the condition of the task does not describe that we need to test the id and I have a test register_addUserWithTheSameLogin_NotOk